.vscode/settings.json

```diff
@@ -0,0 +1,14 @@
+{
```

Comment:
Why are we committing a local editor config file ?

This is specific to each dev (some of them are not even using vscode), I believe this should remain ignored.

Reply:
> Why do we need to commit a local editor config file ?

Is a way of consistently making sure everyone using the most popular editor for WebDev run formatting, import organizing and so on automatically on save.

> This is specific to each dev

Not quite. The `.vscode` folder is a way of sharing configurations, recommended plugins and so on for users of said editor. For example we can commit a `extensions.json` so the devs get recommendations to install extensions like ESLint, Prettier and so on. This is what we do with Astro's UI in Astronomer.

One way we can make it shared but still locally customizable, is to add it back to `.gitignore` and instead add a script that scaffolds this configuration when installing dependencies. That way by default we all get this settings, but we can override them if needed. This way we turn them into the "default recommended settings" instead of the "mandatory settings" for VSCode users. This is what I had set up for my team in VMWare.

Either way we go, I would recommend to keep it because it actually makes DX way better. I would even recommend to also add the `extensions.json` file I mentioned above as well.

Reply:
My concern is people with start forgetting about that and push a whole bunch of changes to `.vscode` without noticing it. (The reason it's ignore in the first place I believe).

We are far from ready to have an official `.vscode` config that works for airflow `back/front/etc.` and I'm not sure we want to push one IDE over another, the majority of people that works on airflow are working on the python side and use pycharm / something else. So if we are to support 1 IDE it's most likely to be the JetBrains suite i.e pycharm/webstorm. (which I don't find great for front-end).

I mean we can I'm sure it will bring a lot of interest and most likely delay this PR. As this is not necessary to the ESLint config, I would recommend to open a separate PR.

Reply:
Good point! I'll revert the `.vscode` changes for now!

Reply:
Done 🎉